eth_utils/applicators.py

```diff
@@ -0,0 +1,79 @@
+from cytoolz.functoolz import (
+    curry,
+    compose,
+)
+
+from .functional import (
+    to_list,
+    to_dict,
+)
+
+
+@curry
+@to_list
```

Comment:
Can we do `to_tuple` cause immutable?!

Reply:
At least in web3, we are counting on this being a list in a number of places (the first one to pop up is the `params` in the middleware is expected to be a list a lot).

Maybe the least surprising solution is to replicate the value passed in. Like `@return_arg_type(2)`.

That seems clean:
- tuple in -> tuple out
- list in -> list out

Reply:
Some quick grepping finds one place where list type is expected.

- `web3/middleware/cache.py:289:    elif params[:1] == ['latest']:`


Even if there are more, they should all be easily convertible to work with tuples or lists.  Seems cleaner to just be indifferent to whether it's a `list/tuple` than to maintain input/output type, but I could be convinced that it's more appropriate to maintain the input/output type as well assuming the code to do so isn't to messy (which I doubt it'll be).